scripts/install-nix-from-closure.sh

```diff
@@ -65,6 +65,21 @@ elif [ "x${1:-}" != "x" ]; then
     exit
 fi
 
+if [ "$(uname -s)" = "Darwin" ]; then
+    canondest="/opt/nix"
+    if ! grep -q '^nix' /etc/synthetic.conf 2> /dev/null; then
```

Comment:
This test is going to fail in two different scenarios:

1. The user has a line like `nixfoo`.
2. The user has already configured this with a bare `nix`, for the filesystem approach.

For the second case we may need to bail and ask the user to clean up their filesystem approach manually first. For the first, we can just tweak the regex. Something like

```bash
if grep -q '^nix$' /etc/synthetic.conf 2>/dev/null; then
  echo "/etc/synthetic.conf contains a bare 'nix'; did you previously set up Nix using a separate filesystem?" >&2
  echo "Please clean up the separate filesystem and remove the bare 'nix' from /etc/synthetic.conf and then re-run the installer." >&2
  exit 1
elif ! grep -q '^nix\t' /etc/synthetic.conf 2>/dev/null; then
  echo "configuring /nix in synthetic.conf" >&2
  …
```